File: CSharp/Contests2025/ABC406/C.cs
class C
{
	static int[] Read() => Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
	static void Main() => Console.WriteLine(Solve());
	static object Solve()
	{
		var n = int.Parse(Console.ReadLine());
		var p = Read();

		var r = 0L;
		var cs = Enumerable.Range(0, n - 1)
			.GroupCountsBySeq(i => p[i] < p[i + 1])
			.Where(p => p.Key)
			.Select(p => p.Value)
			.ToArray();

		for (int i = 1; i < cs.Length; i++)
			r += (long)cs[i - 1] * cs[i];
		return r;
	}
}

static class GE
{
	public static IEnumerable<KeyValuePair<TK, int>> GroupCountsBySeq<TS, TK>(this IEnumerable<TS> source, Func<TS, TK> toKey)
	{
		var c = EqualityComparer<TK>.Default;
		TK k = default(TK), kt;
		var count = 0;

		foreach (var o in source)
		{
			if (!c.Equals(k, kt = toKey(o)))
			{
				if (count > 0) yield return new KeyValuePair<TK, int>(k, count);
				k = kt;
				count = 0;
			}
			++count;
		}
		if (count > 0) yield return new KeyValuePair<TK, int>(k, count);
	}
}

File: Java/Contests2025/ABC406/src/C.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;

public class C {
	static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

	static int[] read() throws IOException {
		return Arrays.stream(in.readLine().split(" ")).mapToInt(s -> Integer.parseInt(s)).toArray();
	}

	public static void main(String[] args) throws IOException {
		System.out.println(solve());
	}

	static Object solve() throws IOException {
		var n = Integer.parseInt(in.readLine());
		var p = read();

		var cs = new ArrayList<Integer>();
		var t = 0;
		for (int i = 1; i < n; i++) {
			if (p[i - 1] < p[i]) {
				t++;
			} else {
				if (t != 0) {
					cs.add(t);
					t = 0;
				}
			}
		}
		if (t != 0) {
			cs.add(t);
		}

		var r = 0L;
		for (int i = 1; i < cs.size(); i++)
			r += (long) cs.get(i - 1) * cs.get(i);
		return r;
	}
}
